src/analyze/analyze.c

```diff
@@ -2154,15 +2156,44 @@ static int do_verify(int argc, char *argv[], void *userdata) {
 
 static int do_security(int argc, char *argv[], void *userdata) {
         _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+        _cleanup_(json_variant_unrefp) JsonVariant *policy = NULL;
         int r;
+        unsigned line, column;
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
                 return bus_log_connect_error(r);
 
         (void) pager_open(arg_pager_flags);
 
-        return analyze_security(bus, strv_skip(argv, 1), arg_scope, arg_man, arg_generators, arg_offline, arg_threshold, arg_root, 0);
+        if (arg_security_policy) {
+                r = json_parse_file(/*f=*/ NULL, arg_security_policy, /*flags=*/ 0, &policy, &line, &column);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to parse '%s' at %u:%u: %m", arg_security_policy, line, column);
+        }
+        else {
+                _cleanup_fclose_ FILE *f = NULL;
+
+                r = search_and_fopen_nulstr("systemd-analyze-security.policy", "re", /*root=*/ NULL, CONF_PATHS_NULSTR("systemd"), &f, /*ret_path=*/ NULL);
```

Comment:
maybe actually pass the last argument, so that you get the filename returned, then use that as second argument to json_parse_file(). In that case json_parse_file() will still read the data off the FILE* that is passed as first arg, but can use the filename to generat besser log messages.